voxblox/include/voxblox/mesh/mesh_integrator.h

```diff
@@ -276,26 +276,18 @@ class MeshIntegrator {
     for (size_t i = 0; i < mesh->vertices.size(); i++) {
       const Point& vertex = mesh->vertices[i];
       VoxelIndex voxel_index = block.computeVoxelIndexFromCoordinates(vertex);
-      const Point voxel_center =
-          block.computeCoordinatesFromVoxelIndex(voxel_index);
-
-      // Should be within half a voxel of the voxel center in all dimensions, or
-      // it belongs in the other one.
-      const Point dist_from_center = vertex - voxel_center;
-      for (unsigned int j = 0; j < 3; ++j) {
-        if (std::abs(dist_from_center(j)) > voxel_size_ / 2.0) {
-          voxel_index(j) += signum(dist_from_center(j));
-        }
-      }
-
       if (block.isValidVoxelIndex(voxel_index)) {
-        mesh->colors[i] = block.getVoxelByVoxelIndex(voxel_index).color;
+        const TsdfVoxel& voxel = block.getVoxelByVoxelIndex(voxel_index);
+
+        if (voxel.weight >= config_.min_weight) {
+          mesh->colors[i] = voxel.color;
+        }
       } else {
-        // Get the nearest block.
         const Block<TsdfVoxel>::ConstPtr neighbor_block =
```

Comment:
Now that the offset part is removed isn't this if else statement trying exactly the same thing twice in a row?